src/super_gradients/training/models/kd_models/kd_module.py

```diff
@@ -0,0 +1,62 @@
+from super_gradients.training.models.sg_module import SgModule
+import torch
+from super_gradients.training.utils.utils import HpmStruct
+
+
+class KDOutput:
+    def __init__(self, student_output: torch.Tensor = None, teacher_output: torch.Tensor = None):
+        self.student_output = student_output
+        self.teacher_output = teacher_output
+
+
+class KDModule(SgModule):
+    """
+    KDModule
+
+    class implementing Knowledge Distillation logic as an SgModule
+
+    attributes:
+        student: torch.nn.Module - the student model
+        teacher: torch.nn.Module- the teacher model
+        freeze_teacher_eval_mode: bool- whether to freeze self.teacher at eval mode regardless of self.train(mode)
```

Comment:
why would someone want to train the teacher? when will this be set to True?

Reply:
freeze_teacher_eval_mode control's whether self.teacher should also transfer to train=True, but that doesn't change the fact the teacher won't be trained (it just controls dropout and bn statistics).
Essentially what this flag means is whether the teacher should have dropout and running bn stats on when training the student or not.

Reply:
This is not clear from the name or documentation. if the difference is between no_grad and eval() than perhaps call the flag "run_teacher_on_eval" or something of that sort and explain in the doc. "Freeze" IMO gives a different connotation. 
and if the teacher will run either with no_grad or eval why do you set all of its params with require_grad=False?

Reply:
When setting the teacher's parameters with requires_grad=False we reduce the network's number of parameters and memory consumption ( https://stackoverflow.com/questions/63785319/pytorch-torch-no-grad-versus-requires-grad-false)
It is true that we don't need to wrap then the teacher's forward with no_grad(), that ill fix now.

Reply:
fixed